lib/src/widgets/chat/typing/ds_dot_animation.widget.dart

```diff
@@ -0,0 +1,150 @@
+class DSDotAnimation extends StatefulWidget {
+  final double size;
+  final Color color;
+  final int numberOfDots;
+  final double innerPadding;
+  final Duration animationTime;
+  final double upLevelAnimation;
+
+  const DSDotAnimation({
+    Key? key,
+    this.numberOfDots = 3,
+    this.size = 10,
+    this.innerPadding = 2.5,
+    this.animationTime = const Duration(milliseconds: 160),
+    this.color = Colors.black,
+    this.upLevelAnimation = -9,
+  }) : super(key: key);
+
+  @override
+  State<DSDotAnimation> createState() => _DSDotAnimationState();
+}
+
+class _DSDotAnimationState extends State<DSDotAnimation>
+    with TickerProviderStateMixin {
+  List<AnimationController>? _animationControllers;
+
+  final List<Animation<double>> _animations = [];
+
+  @override
+  void initState() {
+    super.initState();
+    _initAnimation();
+  }
+
+  @override
+  void dispose() {
+    for (var controller in _animationControllers!) {
+      controller.dispose();
+    }
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Directionality(
+      textDirection: TextDirection.ltr,
+      child: Center(
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: List.generate(
+            widget.numberOfDots,
+            (index) {
+              return DotAnimated(
+                animationControllers: _animationControllers,
+                widget: widget,
+                animations: _animations,
+                index: index,
+              );
+            },
+          ).toList(),
+        ),
+      ),
+    );
+  }
+
+  /// Controls the animation of the dots
+  void _initAnimation() {
+    _animationControllers = List.generate(
+      widget.numberOfDots,
+      (index) {
+        return AnimationController(
+          vsync: this,
+          duration: widget.animationTime,
+        );
+      },
+    ).toList();
+
+    for (int i = 0; i < widget.numberOfDots; i++) {
+      _animations.add(Tween<double>(begin: 0, end: widget.upLevelAnimation)
+          .animate(_animationControllers![i]));
+    }
+
+    for (int i = 0; i < widget.numberOfDots; i++) {
+      _animationControllers![i].addStatusListener(
+        (status) {
+          if (status == AnimationStatus.forward) {
+            if (i != widget.numberOfDots - 1) {
+              Future.delayed(const Duration(milliseconds: 120))
+                  .then((value) => _animationControllers![i + 1].forward());
+            }
+          } else if (status == AnimationStatus.completed) {
+            _animationControllers![i].reverse();
+          }
+          if (i == widget.numberOfDots - 1 &&
+              status == AnimationStatus.dismissed) {
+            _animationControllers![0].forward();
+          }
+        },
+      );
+    }
+    _animationControllers!.first.forward();
+  }
+}
+
+/// Animated dot to be included in the dot list
+class DotAnimated extends StatelessWidget {
```

Comment:
This widget might be reused. So, I'd change his name to DSAnimatedDot to match Flutter's naming convention in animated widgets and create an exclusive file for it.